includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -82,13 +83,69 @@ public function get_selector_conversion_mapping() {
 		];
 	}
 
+	/**
+	 * Mark node and it's child as being PX-verified.
+	 *
+	 * @param DOMNode $node Node.
+	 *
+	 * @return void
+	 */
+	private static function mark_node_as_px_verified_recursively( DOMNode $node ) {
+
+		ValidationExemption::mark_node_as_px_verified( $node );
+
+		if ( $node->hasChildNodes() ) {
+			foreach ( $node->childNodes as $child_node ) {
+				self::mark_node_as_px_verified_recursively( $child_node );
+			}
+		}
+	}
+
+	/**
+	 * Convert picture element into image element or mark as px verified.
+	 *
+	 * @return void
+	 */
+	protected function process_picture_elements() {
+
+		$nodes     = $this->dom->getElementsByTagName( Tag::PICTURE );
+		$num_nodes = $nodes->length;
+
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $index = $num_nodes - 1; $index >= 0; $index-- ) {
+
+			/** @var Element $picture_node */
+			$picture_node = $nodes->item( $index );
+
+			if ( true === $this->args['native_img_used'] ) {
+				self::mark_node_as_px_verified_recursively( $picture_node );
```

Comment:
To be safe, instead of marking all elements as being PX-verified, I suggest only marking `source` and `img` children as such. So this could instead be:

```suggestion
				ValidationExemption::mark_node_as_px_verified( $picture_node );
				foreach ( $this->dom->query( './source | ./img', $picture_node ) as $child_element ) {
		ValidationExemption::mark_node_as_px_verified( $child_element );		
				}
```